pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -1408,6 +1414,18 @@ func (c *Cluster) startMon(m *monConfig, schedule *controller.MonScheduleInfo) e
 	return nil
 }
 
+func isMonIPUpdateRequiredForHostNetwork(mon string, hostNetworkEnabled, isUsingHostNetwork bool) bool {
```

Comment:
I have concerns that the caller could be modified in the future to use `spec.network.hostNetwork` instead of `spec.network.IsHost()`, which would cause incorrect detection for some users. Please change this function signature to take the `c.spec.network` structure, and add unit tests that ensure correct behavior regardless of whether users choose `network.provider == 'host'` or `network.hostnetwork == true`.

Reply:
good point.